code/modules/medical/genetics/bioEffects/beneficial.dm

```diff
@@ -227,14 +227,17 @@
 	degrade_to = "mutagenic_field"
 	icon_state  = "regen"
 
-	OnLife()
+	OnLife(var/mult)
 		if(..()) return
 		var/mob/living/L = owner
-		L.HealDamage("All", heal_per_tick, heal_per_tick)
-		if (rand(1,regrow_prob) == 1 && ishuman(L))
-			var/mob/living/carbon/human/H = L
-			if (H.limbs)
-				H.limbs.mend(1)
+		L.HealDamage("All", heal_per_tick * mult, heal_per_tick * mult)
+		var/roundedmult
+		for (roundedmult = round(mult), roundedmult > 0, roundedmult --)
```

Comment:
Consider keep track of remainder of rounding, that way eventually the fractional delay will add up.  This does help resolve long pauses in OnLife though.